src/Profiles.ts

```diff
@@ -446,141 +439,29 @@ export class Profiles {
-                        const updValue = await vscode.window.showInputBox(options);
-                        if (!Number.isNaN(Number(updValue))) {
-                            updSchemaValues[value] = Number(updValue);
-                            } else {
-                                if (schema[value].optionDefinition.hasOwnProperty("defaultValue")){
-                                    updSchemaValues[value] = schema[value].optionDefinition.defaultValue;
-                                } else {
-                                    break;
-                                }
-                            }
-                        break;
-                    case "boolean" :
-                        let updIsTrue: boolean;
-                        updIsTrue = await this.boolInfo(value, schema);
-                        if (updIsTrue === undefined) {
-                            vscode.window.showInformationMessage(localize("createNewConnection.booleanValue",
-                            "Operation Cancelled"));
-                            return undefined;
-                        }
-                        updSchemaValues[value] = updIsTrue;
-                        break;
-                    default :
-                        options = await this.optionsValue(value, schema, editSession[value]);
-                        const updDefValue = await vscode.window.showInputBox(options);
-                        if (updDefValue === "") {
-                            break;
-                        }
-                        updSchemaValues[value] = updDefValue;
-                        break;
-                }
-            }
-        }
-
-        try {
-            const updSession = await zowe.ZosmfSession.createBasicZosmfSession(updSchemaValues);
-            updSchemaValues.base64EncodedAuth = updSession.ISession.base64EncodedAuth;
-            await this.updateProfile({profile: updSchemaValues, name: profileName, type: profileLoaded.type});
-            vscode.window.showInformationMessage(localize("editConnection.success", "Profile was successfully updated"));
-
-            return updSchemaValues;
-
-        } catch (error) {
-            await errorHandling(error.message);
-        }
+        const newProfile = await this.updateProfile({ profile: profileLoaded.profile, name: profileName, type: profileLoaded.type });
+        vscode.window.showInformationMessage(localize("editConnection.success", "Profile was successfully updated"));
+        return newProfile;
     }
 
     public async getProfileType(): Promise<string> {
         let profileType: string;
         const profTypes = ZoweExplorerApiRegister.getInstance().registeredApiTypes();
         const typeOptions = Array.from(profTypes);
-        if (typeOptions.length === 1 && typeOptions[0] === "zosmf") {
-            profileType = typeOptions[0];
-        } else {
+        if (typeOptions.length === 1 && typeOptions[0] === "zosmf") { profileType = typeOptions[0]; }
```

Comment:
hard code "zosmf" here

Reply:
This is on master branch as well, it is supposed to select zosmf if that is the only option.